rts/lra/lra-client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java

```diff
@@ -739,16 +745,18 @@ private void endLRA(URI lra, boolean confirm) throws WebApplicationException {
         try {
             response = getTarget().path(confirmUrl).request().put();
 
-            if (isUnexpectedResponseStatus(response, Response.Status.OK, Response.Status.ACCEPTED, Response.Status.NOT_FOUND)) {
+            if (isUnexpectedResponseStatus(response, Response.Status.OK, Response.Status.ACCEPTED, NOT_FOUND)) {
                 LRALogger.i18NLogger.error_lraTerminationUnexpectedStatus(response.getStatus(), response.toString());
                 throwGenericLRAException(lra, INTERNAL_SERVER_ERROR.getStatusCode(),
                         "LRA finished with an unexpected status code: " + response.getStatus());
             }
 
-            if (response.getStatus() == Response.Status.NOT_FOUND.getStatusCode()) {
-                LRALogger.logger.infof("Could not %s LRA '%s': coordinator '%s' responded with status '%s'",
-                        confirm ? "close" : "compensate", lra, base, Response.Status.NOT_FOUND.getReasonPhrase());
-                throw new NotFoundException(lra.toASCIIString());
+            if (response.getStatus() == NOT_FOUND.getStatusCode()) {
+                String errorMsg = String.format("Could not %s LRA '%s': coordinator '%s' responded with status '%s'",
```

Comment:
If you use `LRALogger.logger.infof` then you can avoid this string parsing if the level is not info.